lib/entice/logic/MapInstance.ex

```diff
@@ -0,0 +1,67 @@
+defmodule Entice.Logic.MapInstance do
+  alias Entice.Entity
+  alias Entice.Logic.MapInstance
+
+  defstruct(
+    players: [],
+    npcs: [],
+    map: nil)
+
+  def register(entity, map),
+  do: Entity.put_behaviour(entity, MapInstance.Behaviour, map)
+
+  def unregister(entity),
+  do: Entity.remove_behaviour(entity, MapInstance.Behaviour)
+
```

Comment:
we'll need to add the api for joining as a player again here, and the one for adding an npc :)